kuflow-engine-client-activity-impl-s3/src/main/java/com/kuflow/engine/client/activity/s3/S3ActivitiesImpl.java

```diff
@@ -46,9 +47,21 @@ public S3ActivitiesImpl(TaskApi taskApi, S3Client s3Client, S3ActivitiesProperti
     @Override
     public CopyTaskElementFilesResponseResource copyTaskElementFiles(@Nonnull CopyTaskElementFilesRequestResource request) {
         TaskResource task = this.taskApi.retrieveTask(request.getSourceTaskId());
+        if (task.getElementValues() == null) {
+            throw ApplicationFailure.newNonRetryableFailure("ElementValues is empty", "KuFlowActivities.validation");
+        }
```

Comment:
Is more friendly avoid to launch a exception, only return the results with a empty list and no fail

Reply:
But if we do that, the user can think that the process went well, which is not the case, because the `copyTaskElementFiles` method does nothing.

Reply:
If it does but since there is nothing, it does not copy. It works well.
If not, the API client has to do defensive programming, check before calling the activity if it has elements. And as a temporary "failed" process can imply that this process is canceled and the user loses all the work of it, it seems to me a better option to fail as little as possible.